marklogic-data-hub-central/README.md

```diff
@@ -6,27 +6,72 @@ Boot middle tier instead of a Node middle tier
 1. For production, you want to deploy an executable Spring Boot jar that includes all of the static UI resources
 
 
-## Using this during development
+# Developing Data Hub Central
 
 During development, you'll run two separate processes - the Spring Boot middle tier (which defaults to Tomcat), and 
-the UI. Node is still used for building and running the UI. 
-
-To run the UI, first go into the ./marklogic-data-hub-central/ui directory:
-
-    cd marklogic-data-hub-central/ui
-
-The first time you run the UI, and any time after you've made changes or pulled down changes, you'll need to run 
-the following task to pull down all of the UI dependencies:
-
-    npm install
+the UI. Node is still used for building and running the UI. In addition, you'll also need to have a Data Hub instance 
+that you can talk to. Running each of these processes is described below. 
+
+## Running a Data Hub instance
+
+If you're developing locally and not connecting to a Data Hub instance elsewhere - such as hosted by DHS - you'll need 
+to have a Data Hub instance running locally that you can connect to. You'll typically want to build this instance off 
+the branch that you're working on so that it has the latest changes from the develop branch. 
+
+The CONTRIBUTING.md for this repository discusses how to do this in the "Testing changes to the Data Hub Gradle plugin" section. 
+This process is also explained in detail below for convenience when working on Data Hub Central. 
+
+First, run this command from the root directory in this repository to publish the Data Hub Gradle plugin to your local 
+Maven repository (defaults to ~/.m2/repository, but you won't need to access that directly):
+
+    ./gradlew publishToMavenLocal -PskipWeb=
+
+Every time you pull in new changes to this repository, you should run that command to ensure you are working with the 
+latest change to Data Hub. Also, the "skipWeb" property is included to avoid building the QuickStart application in 
+the "./web" directory, which can require a couple minutes to complete. 
+
+Now, you just need to make a build.gradle file that uses this version of the Data Hub Gradle plugin that you published. 
+For example, if you're working on the develop branch for the 5.3.0 release, then when you publish the DH Gradle plugin, 
```

Comment:
How can a developer confirm the version they are suppose to use? Is there a way to inspect the plugin mentioned previously? If a developer is just starting by going to the GitHub project page, how do they know what release they are working on? Where does "5.3" (or whatever the developer is supposed to use) come from?